src/ServiceInsight/SequenceDiagram/ModelCreator.cs

```diff
@@ -98,6 +99,19 @@ void Initialize()
 
                 sendingHandler.Out = sendingHandler.Out.Concat(new[] { arrow }).OrderBy(a => a).ToList();
             }
+
+            ReOrderHandlers(firstOrderHandlers);
+        }
+
+        private void ReOrderHandlers(IList<Handler> orderedHandlers)
```

Comment:
I was thinking the same way @distantcam. I think maybe `Initialize` is not a good name, and we should split that, not adding to the problem. It is already too large to comprehend.